### PR TITLE
Update dependencies to Kafka 2.2.0

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

After adding support for Kafka 2.2 as a version we run, this PR updates the dependencies / Kafka clients used in our applications to 2.2.0 as well. It also updates some related dependencies such as Scala.

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests tags=acceptance,regression

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=LogSettingsST,KafkaST tags=acceptance,regression

File: pom.xml [hunk 1]
```diff
@@ -83,9 +83,9 @@
         <valid4j.version>1.1</valid4j.version>
         <fasterxml.jackson.version>2.9.8</fasterxml.jackson.version>
         <fasterxml.jackson-annotations.version>2.9.8</fasterxml.jackson-annotations.version>
-        <kafka.version>2.1.1</kafka.version>
+        <kafka.version>2.2.0</kafka.version>
         <zkclient.version>0.11</zkclient.version>
-        <scala-library.version>2.12.7</scala-library.version>
+        <scala-library.version>2.12.0</scala-library.version>
```
Comment (ppatierno) [inline pom.xml]: why this downgrade?

Comment (scholzj) [reply pom.xml]: Good catch. Thanks ... 8 looks almost like 0 :-)

**Pull request merged**